src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -93,9 +93,9 @@ public static PackageSpec GetPackageSpec(JObject rawPackageSpec, string name, st
             {
                 try
                 {
-                    bool hasVersionSnapshot;
-                    packageSpec.Version = SpecifySnapshot(version.Value<string>(), snapshotValue, out hasVersionSnapshot);
-                    packageSpec.HasVersionSnapshot = hasVersionSnapshot;
+                    var versionString = version.Value<string>();
+                    packageSpec.HasVersionSnapshot = PackageSpecUtility.IsSnapshotVersion(versionString);
```

Comment:
We still need to support pack for project.json right? That is the only place this is used today. .NETCore projects will fail before getting here.